ci_scripts/assert_news.py

```diff
@@ -1,86 +1,153 @@
+            file_content = fh.read()
+        if not file_content:
+            raise ValueError(f'Empty news file `{self.news_file_path}`')
+        if len(file_content.splitlines()) > 1:
+            raise ValueError('News file must only contain 1-line sentence')
+
+    def verify(self):
+        """
+        Verifies news file follows standards
+        """
+        logger.info('Verifying %s' % self.news_file_path)
+        self._verify_news_file_name()
+        self._verify_news_file_content()
+
+
+class NewsFileDiscoverer(object):
+    """
+    Checks that all new PRs comprise a news file and that such files follow the standard
+    """
+
+    def __init__(self):
+        self.git = GitWrapper()
+        self.current_branch = self.git.get_current_branch()
+        self.master_branch = MASTER_BRANCH
+
+    def find_news_file(self):
+        """
+       Determines a list of all the news files which were added as part of the PR.
+        :return: list of introduced news files
+        """
+
+        if not os.path.exists(NEWS_DIR):
+            NotADirectoryError(
+                f'News files directory was not specified and default path `{NEWS_DIR}` does not exist'
+            )
+        logger.info(
+            f':: Looking for news files in `{NEWS_DIR}` [{self.current_branch}]'
+        )
+        self.git.checkout(self.current_branch)
+        self.git.set_upstream_branch(self.current_branch)
+        self.git.pull()
+        current_commit_hash = self.git.get_commit_hash()
+        # Delete `master` as it may already exist and be set to the one of interest
+        if self.git.branch_exists(self.master_branch):
+            self.git.delete_branch(self.master_branch)
+        self.git.checkout(self.master_branch)
+        self.git.set_upstream_branch(self.master_branch)
+        self.git.pull()
+        master_branch_commit_hash = self.git.get_commit_hash()
+        self.git.checkout(self.current_branch)
+        added_news = self.git.get_changes_list(
+            'a', self.git.get_branch_point(
+                master_branch_commit_hash, current_commit_hash),
+            current_commit_hash,
+            os.path.relpath(NEWS_DIR,
+                            os.path.commonprefix([NEWS_DIR, PROJECT_ROOT]))
+        )
+        extension_to_exclude = ['.toml', '.rst']
+        return [path for path in added_news if
+                len([ex for ex in extension_to_exclude if ex in path]) == 0]
+
+    def verify(self):
+        """
+        Checks that news files were added in the current branch as part of the PR's changes
+        The files are then individually checked in order to ensure they follow the standard in terms of naming and content.
+        """
+        if self.current_branch in [MASTER_BRANCH, BETA_BRANCH] or re.search(
+                RELEASE_BRANCH_PATTERN, self.current_branch):
+            logger.info(
+                f'No need for news file on branch [{self.current_branch}]'

```

Comment:
Worth saying why there is no need for a news file, new readers may not have the background.